src/ior.c

```diff
@@ -1744,33 +1744,50 @@ static IOR_offset_t *GetOffsetArraySequential(IOR_param_t * test,
         offsetArray[offsets] = -1;      /* set last offset with -1 */
 
         /* fill with offsets */
-        for (i = 0; i < test->segmentCount; i++) {
-                for (j = 0; j < (test->blockSize / test->transferSize); j++) {
-                        offsetArray[k] = j * test->transferSize;
-                        if (test->filePerProc) {
-                                offsetArray[k] += i * test->blockSize;
-                        } else {
-                                offsetArray[k] +=
-                                        (i * test->numTasks * test->blockSize)
-                                        + (pretendRank * test->blockSize);
+        IOR_offset_t next_offset, incrament, trips;
+        /* special cases for filePerProc and transferSize=blockSize */
+        if (test->filePerProc) {
+                next_offset = 0;
+                incrament = test->transferSize;
+                trips = test->segmentCount * (test->blockSize / test->transferSize);
+                for (i = 0; i < trips; i++) {
+                        offsetArray[i] = next_offset;
+                        next_offset += incrament;
+                }
+        } else {
+                if (test->transferSize == test->blockSize) {
+                        next_offset = pretendRank * test->blockSize;
+                        incrament = test->numTasks * test->blockSize;
+                        trips = test->segmentCount;
+                        for (i = 0; i < trips; i++) {
+                                offsetArray[i] = next_offset;
+                                next_offset += incrament;
+                        }
+                } else {        /* usual case is segmentCount=1 */
```

Comment:
Do we need an explicit test for segmentCount==1?  What happens here if it isn't?